spec/requests/api/provision_requests_spec.rb

```diff
@@ -95,6 +95,155 @@
+    end
+    let(:ems) { FactoryGirl.create(:ems_amazon_with_authentication) }
+    let(:template) do
+      FactoryGirl.create(:template_amazon, :name => "template1", :ext_management_system => ems)
+    end
+    let(:flavor) do
+      FactoryGirl.create(:flavor_amazon, :ems_id => ems.id, :name => 't2.small', :cloud_subnet_required => true)
+    end
+    let(:az)             { FactoryGirl.create(:availability_zone_amazon, :ems_id => ems.id) }
+    let(:cloud_network1) { FactoryGirl.create(:cloud_network_amazon, :ems_id => ems.network_manager.id, :enabled => true) }
+    let(:cloud_subnet1) do
+      FactoryGirl.create(:cloud_subnet, :ems_id => ems.id, :cloud_network => cloud_network1, :availability_zone => az)
+    end
+    let(:security_group1) do
+      FactoryGirl.create(:security_group_amazon, :name => "sgn_1", :ext_management_system => ems,
+                         :cloud_network => cloud_network1)
+    end
+    let(:floating_ip1) do
+      FactoryGirl.create(:floating_ip_amazon, :cloud_network_only => true, :ems_id => ems.network_manager.id,
+                         :cloud_network => cloud_network1)
+    end
+
+    let(:provreq_body) do
+      {
+        "template_fields" => {"guid" => template.guid},
+        "requester"       => {
+          "owner_first_name" => "John",
+          "owner_last_name"  => "Doe",
+          "owner_email"      => "user@example.com"
+        }
+      }
+    end
+
+    let(:expected_provreq_attributes) { %w(id options) }
+
+    let(:expected_provreq_hash) do
+      {
+        "userid"         => api_config(:user),
+        "requester_name" => api_config(:user_name),
+        "approval_state" => "pending_approval",
+        "type"           => "MiqProvisionRequest",
+        "request_type"   => "template",
+        "message"        => /Provisioning/i,
+        "status"         => "Ok"
+      }
+    end
+
+    it "supports manual placement" do
+      api_basic_authorize collection_action_identifier(:provision_requests, :create)
+
+      body = provreq_body.merge(
+        "vm_fields" => {
+          "vm_name"                     => "api_test_aws",
+          "instance_type"               => flavor.id,
+          "placement_auto"              => false,
+          "placement_availability_zone" => az.id,
+          "cloud_network"               => cloud_network1.id,
+          "cloud_subnet"                => cloud_subnet1.id,
+          "security_groups"             => security_group1.id,
+          "floating_ip_address"         => floating_ip1.id
+        }
+      )
+
+      run_post(provision_requests_url, body)
+
+      expect_request_success
+      expect_result_resources_to_include_keys("results", expected_provreq_attributes)
+      expect_results_to_match_hash("results", [expected_provreq_hash])
+
+      expect(response_hash["results"].first).to include(
+        "options" => include(
+          "placement_auto"              => include(false),
```

Comment:
Ah OK, thanks :thumbsup:
Would it be possible to use arrays for those instead of the `include` matcher? If not, you can use `a_collection_including` when composing matchers to give a better clue as to what kind of thing it is. Similarly, you can use `a_hash_including` if it's a hash, as above.

Actually, having said that, it might be better to do something like this, if you can't determine one or more items in the array:

```ruby
[anything, false] # `anything` will match against any value
```